pkg/stream/redis.go

```diff
@@ -0,0 +1,42 @@
+package stream
+
+import (
+	"context"
+
+	"github.com/go-redis/redis/v8"
+	"github.com/lthibault/log"
+)
+
+type RedisPubsub struct {
+	Redis  *redis.Client
+	Logger log.Logger
+}
+
+func (r *RedisPubsub) Publish(ctx context.Context, topic string, data []byte) error {
+	return r.Redis.Publish(ctx, topic, data).Err()
+}
+
+func (r *RedisPubsub) Subscribe(ctx context.Context, topic string) (chan []byte, error) {
```

Comment:
As you never return error here. any chance to consider 
```suggestion
func (r *RedisPubsub) Subscribe(ctx context.Context, topic string, sub chan []byte) {
```
and then just  `go r.Subscribe( .. ) ` somewhere else? 
It may allow then the creation and termination (closing) of the channel outside of this goroutine.  Unless it's essential to do it right here